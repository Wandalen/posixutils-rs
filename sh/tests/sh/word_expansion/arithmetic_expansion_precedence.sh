echo $((+1 + 2))
echo $((-1 + 2))
echo $((2 * +3))
echo $((2 * -3))
echo $((+2 * -3))
echo $((-2 * +3))
echo $((~1 + 2))
echo $((3 * ~1))
echo $((!0 && 2))
echo $((2 + 3 * 4))
echo $((2 * 3 + 4))
echo $((15 / 3 + 2 * 4))
echo $((2 << 2 + 1))
echo $((8 >> 2 * 2))
echo $((5 + 3 < 4 + 5))
echo $((2 * 3 <= 4 << 1))
echo $((2 + 2 > 4 - 1))
echo $((8 - 2 >= 3 * 2))
echo $((2 + 2 == 2 * 2))
echo $((3 * 2 != 5 + 1))
echo $((4 & 5 | 6))
echo $((4 | 5 & 6))
echo $((3 ^ 4 & 5))
echo $((1 && 2 || 0))
echo $((0 || 1 && 0))
echo $((1 + 2 ? 3 * 2 : 4 / 2))
echo $((0 * 2 ? 3 + 2 : 4 << 1))

x=1
echo $((x += 2 * 3))
echo $((x *= 2 + 3))
echo $((x /= 3 << 1))
echo $((x %= 2 + 3))
echo $((x <<= 1 + 1))
echo $((x >>= 2 * 1))
echo $((x &= 3 | 4))
echo $((x ^= 2 + 3))
echo $((x |= 4 & 7))