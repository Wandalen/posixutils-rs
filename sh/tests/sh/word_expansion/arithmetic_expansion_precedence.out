3
1
6
-6
-6
-6
0
-6
1
14
10
13
16
0
1
1
1
1
1
0
6
4
7
1
0
6
8
7
35
5
0
0
0
0
5
5
